infra/prover/index.ts

```diff
@@ -0,0 +1,338 @@
+  const brokerTomlPath = config.require('BROKER_TOML_PATH')
+  
+  const ethRpcUrlSecret = new aws.secretsmanager.Secret(`${serviceName}-brokerEthRpc`);
+  const _ethRpcUrlSecretSecretVersion = new aws.secretsmanager.SecretVersion(`${serviceName}-brokerEthRpc`, {
+    secretId: ethRpcUrlSecret.id,
+    secretString: ethRpcUrl,
+  });
+  
+  const privateKeySecret = new aws.secretsmanager.Secret(`${serviceName}-brokerPrivateKey`);
+  const _privateKeySecretVersion = new aws.secretsmanager.SecretVersion(`${serviceName}-privateKeyValue`, {
+    secretId: privateKeySecret.id,
+    secretString: privateKey,
+  });
+
+  const bonsaiSecret = new aws.secretsmanager.Secret(`${serviceName}-brokerBonsaiKey`);
+  const _bonsaiSecretVersion = new aws.secretsmanager.SecretVersion(`${serviceName}-bonsaiKeyValue`, {
+    secretId: bonsaiSecret.id,
+    secretString: bonsaiApiKey,
+  });
+
+  const brokerS3Bucket = new aws.s3.Bucket(serviceName, {
+    bucketPrefix: `boundless-${serviceName}`,
+    tags: {
+      Name: serviceName,
+    },
+  });
+
+  const fileToUpload = new pulumi.asset.FileAsset(brokerTomlPath);
+
+  const bucketObject = new aws.s3.BucketObject(serviceName, {
+    bucket: brokerS3Bucket.id,
+    key: 'broker.toml',
+    source: fileToUpload,
+  });
+
+  const taskRole = new aws.iam.Role(serviceName, {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+      Service: 'ecs-tasks.amazonaws.com',
+    }),
+  });
+
+  const _rolePolicy = new aws.iam.RolePolicy(serviceName, {
+    role: taskRole.id,
+    policy: {
+      Version: '2012-10-17',
+      Statement: [
+        {
+          Effect: 'Allow',
+          Action: ['s3:GetObject', 's3:ListObject', 's3:HeadObject'],
+          Resource: [bucketObject.arn],
+        },
+        {
+          Effect: 'Allow',
+          Action: ['secretsmanager:GetSecretValue', 'ssm:GetParameters'],
+          Resource: [privateKeySecret.arn, bonsaiSecret.arn],
+        },
+      ],
+    },
+  });
+
+  const execRole = new aws.iam.Role(`exec-${serviceName}`, {
+    assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+      Service: 'ecs-tasks.amazonaws.com',
+    }),
+  });
+
+  const _execRolePolicy = new aws.iam.RolePolicy(`exec-${serviceName}`, {
+    role: execRole.id,
+    policy: {
+      Version: '2012-10-17',
+      Statement: [
+        {
+          Effect: 'Allow',
+          Action: [
+            'ecr:GetAuthorizationToken',
+            'ecr:BatchCheckLayerAvailability',
+            'ecr:GetDownloadUrlForLayer',
+            'ecr:BatchGetImage',
+            'logs:CreateLogStream',
+            'logs:PutLogEvents',
+          ],
+          Resource: '*',
+        },
+        {
+          Effect: 'Allow',
+          Action: ['secretsmanager:GetSecretValue', 'ssm:GetParameters'],
+          Resource: [privateKeySecret.arn, bonsaiSecret.arn],
+        },
+      ],
+    },
+  });
+
+  const brokerEcr = new awsx.ecr.Repository(serviceName, {
```

Comment:
One thing I would avoid using `awsx` and stick `aws`. There's many reasons why but the main one is that the `awsx` package doesn't have all the component resource functions that the `aws` sdk has.